go.mod

```diff
@@ -1,15 +1,15 @@
 module github.com/BurntSushi/go-sumtype
 
-go 1.17
+go 1.23.0
```

Comment:
Why is this necessary?

Reply:
@BurntSushi This was updated automatically when running `go get -u golang.org/x/tools`. The latest `x/tools` version, v0.31.0, specifies go1.23.0 as the minimum required version in its [go.mod](https://cs.opensource.google/go/x/tools/+/refs/tags/v0.31.0:go.mod;l=3).

Reply:
Is `0.31.0` the earliest version that fixes the problem you encountered?

I'm just trying to avoid being too eager about requiring a newer Go version. I could easily imagine merging this and then immediately getting complaints that it "no longer works on Go 1.20" or whatever. So I want to do my due diligence first.

Reply:
Thank you for the clarification!
The earliest version that fixes the problem is `0.25.0` which requires a minimum Go version of [`1.22.0`](https://cs.opensource.google/go/x/tools/+/refs/tags/v0.25.0:go.mod;l=3). From `0.25.0` to `0.30.0`, the minimum required Go version remains `1.22.0`. 
So I think updating `golang.org/x/tools` to `0.30.0` is reasonable, as it includes the necessary fixes while still being compatible with Go `1.22.0`. Does that sound correct to you?

Reply:
I think that's fine. Thanks for doing the leg work here.